### PR TITLE
Migrate to aws v2

### LINKED ISSUE #529: [internal/storage] Migrate from `github.com/aws/aws-sdk-go` to `github.com/aws/aws-sdk-go-v2`
**Please describe the feature**
As a developer, I want the code to use the latest version of a module so that we can make use of the new functionality and security options.

**Acceptance criteria**
- [ ] 
- [ ] Tests verifying the changes are added

**Additional context**

**Estimation of size**: small/medium/big

**Estimation of priority**: low/medium/high

### PR DESCRIPTION
**Pull request by jbygdell**
**Related issue(s) and PR(s)**  
This PR closes #529, #530 .

**Description**
This PR Migrates the `sda` code base to V2 of the `aws-sdk`

**How to test**

### CODE REVIEW

**Comment by blankdots** [comment]
does it make sense changes for #378 be done as part of this PR ?

**Comment by jbygdell** [comment]
> does it make sense changes for #378 be done as part of this PR ?

No, I don't think so.

**Review by pontus — APPROVED** [review]

File: sda/cmd/s3inbox/proxy.go [hunk 1]
```diff
@@ -463,32 +457,20 @@ func (p *Proxy) CreateMessageFromRequest(r *http.Request, claims jwt.Token) (Eve
...
-		Bucket:  aws.String(p.s3.Bucket),
-		MaxKeys: aws.Int64(1),
-		Prefix:  aws.String(filePath),
+		Bucket:  &p.s3.Bucket,
+		MaxKeys: aws.Int32(1),
+		Prefix:  &filePath,
 	}
 
-	result, err := svc.ListObjectsV2(input)
+	result, err := client.ListObjectsV2(context.TODO(), input)
```
Comment (blankdots) [inline sda/cmd/s3inbox/proxy.go:471]: why not `context.Background` https://pkg.go.dev/context#Background ?

Comment (jbygdell) [reply sda/cmd/s3inbox/proxy.go:471]: According to the doc, when you need a context but you don't have one (yet) and don't know what to use, use context.TODO().

Comment (blankdots) [reply sda/cmd/s3inbox/proxy.go:471]: ok, i understood those are similar, just some semantic differences. I am fine with other

```
Background()` function provides a root context for use when no other context is available. The `context. TODO()` function serves as a placeholder for when it is unclear which context to use
```
https://reintech.io/term/context-background-and-context-todo-in-go

**Review by blankdots — APPROVED** [review]

**Pull request merged**